extra_tests/snippets/set.py

```diff
@@ -195,7 +195,8 @@ def __hash__(self):
 i = iter(a)
 a.add(4)
 a.remove(4)
-assert next(i) == 1
+with assert_raises(RuntimeError):
```

Comment:
Note on this:

Set behaves the same as dictionaries and detects mutations even when they are balanced, i.e:

```
d = {1: 2, 3: 4}
dit = iter(d)
d[5] = 6
del d[6]
next(dit)  # RuntimeError
```

CPython doesn't detect any of these but openly states in [PEP 3106](https://www.python.org/dev/peps/pep-3106/#id5) that mutations while iterating is undefined behavior:

> As in Python 2.x, mutating a dict while iterating over it using an iterator has an undefined effect and will in most cases raise a `RuntimeError` exception. (This is similar to the guarantees made by the Java Collections Framework.) 

So I'm pretty sure that a `RuntimeError` is the correct approach here.